### PR TITLE
feat: extend SelectAllOption props to enable setting individual itemIds

### PR DESCRIPTION
**Pull request by Lelith**
# Purpose of PR
 This will allow for the MultiSelect SubComponent SelectAllOption to handover an individual itemId to further customize this components experience.

### CODE REVIEW

**Review by cf-remylenoir — APPROVED** [review]
It looks good to me. I was curious to see if the API documentation was reflecting how to use this and realized [we don't have it rendered yet](https://forma-36-git-feat-multiselectselectallitemidasprop.colorfuldemo.com/components/multiselect#props-api-reference). We should keep this in mind for when the component exits the `alpha`.

**Comment by Lelith** [comment]
@cf-remylenoir what do you mean with we do not have it rendered? there is an example here: 
https://forma-36-git-feat-multiselectselectallitemidasprop.colorfuldemo.com/components/multiselect#selectall-option 
and one in the storybook: https://f36-storybook.contentful.com/?path=/story/components-multiselect--with-select-all

File: .changeset/swift-shrimps-sneeze.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@contentful/f36-multiselect': patch
+---
+
+Allows now setting an individual itemId for the SelectAll sellect option
```
Comment (Lelith) [inline .changeset/swift-shrimps-sneeze.md:5]:
```suggestion
Allows now setting an individual itemId for the SelectAllOption component which is a divert from MultiselectOption
```

**Comment by cf-remylenoir** [comment]
> @cf-remylenoir what do you mean with we do not have it rendered? there is an example here: https://forma-36-git-feat-multiselectselectallitemidasprop.colorfuldemo.com/components/multiselect#selectall-option and one in the storybook: https://f36-storybook.contentful.com/?path=/story/components-multiselect--with-select-all

Thanks @Lelith, I meant the actual API prop table 🕵️

**Comment by Lelith** [comment]
@cf-remylenoir ah i think the prop table might have issues with sub-components as SelectAll is only a compound Component and its not a prop on the Multiselect Component itself.

**Pull request merged**